### ISSUE TITLE
DataTable: Empty aria-label generation in multiselect tables 

### ISSUE THREAD

**Issue by Bukama** [primefaces/primefaces#10429]
### Describe the bug

During our improvements / rewrites to make our webapp accessibly and fullfil the requirements of the official German BITV 2.0 test we came along a slightly problem with multi select [data tables](https://www.primefaces.org/showcase/ui/data/datatable/selection.xhtml?jfwid=8db7c).
The tried to achieve that every checkbox has an (hidden) description like "select row with identifier X" so the screen reader (we use NVDA, which is also used in the official test) reads to a visual impaired person what row he selects, when ticking the checkbox.

For this we tried to make use of the PF API attribute `ariaRowLabel` which descriptions sounds to do exactly this. But it seems that it does not as the screenreader keeps quiet, regardless of the value inside. So we used a pass-trough attribute which is picked up correctly by the screenreader, but which is not rendered at the checkbox, but at the `td` tag.

So this code 
```
<p:column 
   headerText="Selecton"
   selectionMode="multiple" 
   pt:aria-label="this is the row with PK [value]"
   style="width:10ch; text-align:left">
</p:column> 
```

generates this outcome (for row with PK value 1) in PF 12.00 (Not yet tested with PF 13, but release notes don't show any changes here and the  sadly has no correspondening example :( )

```
<td 
    role="gridcell" 
    style="width:10ch; text-align:left" class="ui-selection-column" 
    aria-label="this is the row with PK 1">
    <div 
        class="ui-chkbox ui-widget">
        <div 
            id="frm_kleineSucheSuchErgebnisse:row_2:crd_2:tbl_suchErgebnisse_1621_checkbox" 
            role="checkbox" 
            tabindex="0" 
            aria-label="" 
            aria-checked="false" 
            class="ui-chkbox-box ui-widget ui-corner-all ui-state-default"
            >
            <span class="ui-chkbox-icon ui-icon ui-icon-blank ui-c"></span>
        </div>
    </div>
</td>

```

As you can see, this renderes an empty `aria-label` for the "checkbox-div". 

So why do I file an issue, when the screen reader handles it correctly?

Two reasons:

* I / We assume that the empty `aria-label` should have the value the `td` has, where the `td` should probably have none or the same? (Don't know if the same would cause trouble tbh)
* The lighthouse report claims that the checkbox has no aria label and is therefore not accessible
* What is the `ariaRowLabel` for, if not for this?

> ARIA toggle fields do not have accessible names
> 
> When a toggle field doesn't have an accessible name, screen readers announce it with a generic name, making it unusable
> for users who rely on screen readers.
> 
> div#frm_kleineSucheSuchErgebnisse:row_2:crd_2:tbl_suchErgebnisse_1621_checkbox.
> ui-chkbox-box.ui-widget.ui-corner-all.ui-state-default

Further notes
* The lightous report always claimes that error, regardless of usage of a `pt` or the `ariaRowLabel` or none/bith of them
* Using the `pt` makes the screenreader work, not using or only using the `ariaRowLabel` does not

### PrimeFaces edition

Community

### PrimeFaces version

12.0.0

### JSF implementation

MyFaces

### JSF version

2.2

### Java version

8

**Comment by melloware** [comment]
@Bukama  I think I fixed this yesterday with this PR: https://github.com/primefaces/primefaces/pull/10424

Now the Select All Checkbox has proper ARIA labeling of "All items selected" or "All items unselected"

**Comment by Bukama** [comment]
> @Bukama I think I fixed this yesterday with this PR: #10424
> 
> Now the Select All Checkbox has proper ARIA labeling of "All items selected" or "All items unselected"

It's not about the "all items selected", but the checkbox in each individual line / for each individual record (which has [for accessibility] to have an individual description text with which helps the user to identify this specific row)

**Comment by melloware** [comment]
Oh my fault you meant each row.  Let me see what I did in PrimeReact I think I fixed this there.

**Comment by melloware** [comment]
Yep I did fix it in PrimeReact: https://github.com/primefaces/primereact/issues/3263

I will see if I can make the same fixes here.

**Comment by melloware** [comment]
It now makes it `aria-label="Row Selected 1000"` or `aria-label="Row Unselected 1000"`  using the databale `rowKey` field as the unique identifier.

The labels are I8LN using these values.

```js
         "selectRow": "Row Selected",
         "unselectRow": "Row Unselected",
```

So in German 

```js
        "selectRow": "Zeile ausgewählt",
        "unselectRow": "Zeile abgewählt",
```
